### ISSUE TITLE
[Linux] Window manager close button missing in settings dialog

### ISSUE THREAD

**Issue by ainola** [owncloud/client#3713]
The OS already provides a close button. There is no reason for an application to provide its own. All it does is clutter up the application.

**Comment by danimo** [comment]
@jancborchardt?

**Comment by guruz** [comment]
@ainola Which close button? Which OS? etc.

**Comment by phil-davis** [comment]
I will guess the Close button in the bottom right corner of the ownCloud client GUI.
![close](https://cloud.githubusercontent.com/assets/1535615/9576734/61055496-4ffa-11e5-8ea4-3f197ee03247.png)

**Comment by danimo** [comment]
My stance on this: It's a dialog, and as such should always have a button. I agree that we are wasting space on the right, but it could contain e.g. add/delete account or other stuff that is tab-specific.

**Comment by ainola** [comment]
Indeed, I meant the bottom right button. 

If we have a close button widget, why not include a minimize and maximize button as well? It is entirely redundant. Users should look to their window controls to manage windows, not the contents of the window.

**Comment by danimo** [comment]
@ainola It's a matter of expectation. All dialogs have a button to dismiss the content. If you want to change that, I wonder if starting at arbitrary applications is the way to go..

**Comment by ainola** [comment]
If this is a KDE app, perhaps you are right: I don't know what KDE's HIG recommends. Gnome's HIG does not recommend this behavior, though.

"All dialogs have a button to dismiss the content."

Yes, it's that little 'X' button at the top right/left of your window. I wasn't aware that owncloud was considered a dialogue.... it's the main application window. To me, it's as if Firefox decided to allocate the bottom right space with a close widget.

**Comment by danimo** [comment]
@ainola ownCloud doesn't really have a main window. It's supposed to behave like a service, and not bother you at all. If you choose to make changes to check the status, you open the dialog. It's more than a settings dialog, but it has dialog semantics nontheless.

**Comment by ainola** [comment]
If it is supposed to behave like a service and not bother, why is there still no option to remove the status icon? :)

Also, by technicality it does has a main window. Whether that statement belies pure pedantry or not, I'll let you decide. 

I have just noticed that the interface has been getting ever more cluttered and offered up a solution. I see that it was not well received. That's fine, though hopefully in the back of your mind you'll now notice close widgets as the pointless repetition they are :)

Thanks for all your work!

**Comment by danimo** [comment]
> If it is supposed to behave like a service and not bother, why is there still no option to remove the status icon? :)

Because we're not there yet.

**Comment by jancborchardt** [comment]
I totally agree that the Close button is totally redundant. Also talked to @ogoffart @guruz @jturcotte about that yesterday. I’d be in favor of removing it.

**Comment by guruz** [comment]
@ainola Ah, it is in the Linux/Windows implementation, that's why I never saw it. The OS X code is unfortunately different codebase for that dialog. thanks for reporting

**Comment by jancborchardt** [comment]
By the way: At least on Ubuntu with Gnome, the dialog has no X button in the top right. Should have.

**Comment by jboisdequin** [comment]
I'm using Cinnamon on my Ubuntu 14.04 laptop. I am now unable to close the owncloud client window (see attachment). I don't have the 'X' button at the top right/left of the window

![selection_010](https://cloud.githubusercontent.com/assets/791037/10659676/c93d1dbc-78a1-11e5-90e0-319b5f0d5e96.png)

**Comment by jancborchardt** [comment]
@ogoffart @guruz same here with GNOME – the X in the top right is missing. Can we force that?

**Comment by guruz** [comment]
@jboisdequin @jancborchardt Could you guys check in the client settings dialog under "General" which Qt version it is using?

**Comment by jboisdequin** [comment]
I'm using Qt 4.8.6

2015-10-22 14:53 GMT+02:00 Markus Goetz <notifications@github.com>:

> @jboisdequin <https://github.com/jboisdequin> @jancborchardt
> <https://github.com/jancborchardt> Could you guys check in the client
> settings dialog under "General" which Qt version it is using?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/owncloud/client/issues/3713#issuecomment-150210798>.
>



-- 
*Julien BOISDEQUIN*
IT Consultant
Mobile: +32475347237
GPG Key-ID: 775CF837

**Comment by guruz** [comment]
@jancborchardt And you?

@jboisdequin and which client version?

**Comment by jboisdequin** [comment]
@guruz client version v2.0.2

**Comment by danimo** [comment]
tl;dr There is no reliable way of hinting Window managers which buttons to show on Linux. All you can do is plea. We had requests to hide the maximize/minimize buttons but then you need to hint a dialog (which some windows decide not to show an X button for.

Let's simply do what every other sensible application in the world of the desktop does and reintroduce the close button.

**Comment by danimo** [comment]
(Closing instead of ready-to-test because I just verified that the revert works).

**Comment by dragotin** [comment]
Please. Never again we remove desktop standard gui elements.

**Comment by jancborchardt** [comment]
Well, I guess every desktop has their own standard. So we need to introduce a non-standard element as a hack to fix it. If the X would be there, no need for the supposedly »standard« extra Close button. ;)

**Comment by danimo** [comment]
> If the X would be there

But it isn't, and we have to adjust to this reality. And we're certainly not doing per-desktop ifdefs.

**Comment by jturcotte** [comment]
Maybe it only make sense to remove the close button if we use a non-dialog window type. This would then make the settings window show up in the DE's taskbar, so not really sure if it's something we want.

**Comment by dragotin** [comment]
Very easy: No extra complexity and efforts for Close-button Yes/No. We simply keep it.
